src/fastcs_eiger/eiger_parameter.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import Any, Literal
+
+from fastcs.datatypes import Bool, DataType, Float, Int, String
+from pydantic import BaseModel
+
+
+class EigerParameterResponse(BaseModel):
+    access_mode: str
+    allowed_values: Any | None = None
+    value: Any
+    value_type: Literal["float", "int", "bool", "str"]
+
+    @property
+    def fastcs_datatype(self) -> DataType:
+        match self.value_type:
+            case "float":
+                return Float()
+            case "int":
+                return Int()
+            case "bool":
+                return Bool()
+            case "str":
+                return String()
+            case _:
+                raise ValueError(f"Unsupported data type: {self.value_type}")
```

Comment:
Following how this was previously done, I think this should be:
```suggestion
    value_type: Literal["float", "int", "bool", "uint", "string" , "datetime" , "State" , "string[]"]

    @property
    def fastcs_datatype(self) -> DataType:
        match self.value_type:
            case "float":
                return Float()
            case "int" | "uint" :
                return Int()
            case "bool":
                return Bool()
            case "string" | "datetime" | "State" | "string[]":
                return String()
            case _:
                raise ValueError(f"Unsupported data type: {self.value_type}")
```

Which should resolve your initial validation issues.

Reply:
I was initially following the fastcs_odin example for value types, assuming those were the recommended ones. But your suggestion makes a lot of sense, there aren't any validation issues now. I will update the code.